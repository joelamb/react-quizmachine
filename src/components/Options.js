import React from 'react';

const Options = ({ handleSubmit }) => {
  const categories = [
    {
      id: 9,
      category: "General Knowledge",
    },
    {
      id: 12,
      category: "Music",
    },
    {
      id: 17,
      category: "Science & Nature"
    },
    {
      id: 20,
      category: "Mythology",
    },
    {
      id: 27,
      category: "Animals",
    }
  ]

  const difficulties = ["easy", "medium", "hard"];
  const initialId = categories[0].id;
  const initialDifficulty = difficulties[0];

  return (
    <form onSubmit={(e) => {
      e.preventDefault();
      handleSubmit(e.target.categorySelect.value, e.target.difficultySelect.value);
    }}>
      <fieldset>
        <legend>Select a category</legend>
        {categories.map(item => {
          const { id, category } = item;
          return <div key={id}>
            <input
              type="radio"
              name="categorySelect"
              id={category}
              value={id}
              defaultChecked={id == initialId ? true : false} />
            <label htmlFor={category}>{category}</label>
          </div>
        })}
      </fieldset>

      <fieldset>
        <legend>Select a difficulty level</legend>
        {difficulties.map(item => {
          return <div key={item}>
            <input type="radio" name="difficultySelect" id={item} value={item} defaultChecked={item == initialDifficulty ? true : false} />
            <label htmlFor={item}>{item}</label>
          </div>
        })}
      </fieldset>
      <input type="submit" value="Submit" />
    </form>
  )
}

export default Options;