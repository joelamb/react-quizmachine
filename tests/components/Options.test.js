import React from 'react'
import Options from '../../src/components/Options';
import renderer from 'react-test-renderer';

describe('Options', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<Options />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
})