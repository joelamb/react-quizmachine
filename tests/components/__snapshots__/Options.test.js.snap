// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options should render correctly 1`] = `
<form
  className="options"
  onSubmit={[Function]}
>
  <fieldset
    className="options__category"
  >
    <legend>
      Select a category
    </legend>
    <p>
      <label
        htmlFor="Any"
      >
        <input
          defaultChecked={true}
          id="Any"
          name="categorySelect"
          type="radio"
          value={0}
        />
        <span>
          Any
        </span>
      </label>
    </p>
    <p>
      <label
        htmlFor="General Knowledge"
      >
        <input
          defaultChecked={false}
          id="General Knowledge"
          name="categorySelect"
          type="radio"
          value={9}
        />
        <span>
          General Knowledge
        </span>
      </label>
    </p>
    <p>
      <label
        htmlFor="Music"
      >
        <input
          defaultChecked={false}
          id="Music"
          name="categorySelect"
          type="radio"
          value={12}
        />
        <span>
          Music
        </span>
      </label>
    </p>
    <p>
      <label
        htmlFor="Science & Nature"
      >
        <input
          defaultChecked={false}
          id="Science & Nature"
          name="categorySelect"
          type="radio"
          value={17}
        />
        <span>
          Science & Nature
        </span>
      </label>
    </p>
    <p>
      <label
        htmlFor="Mythology"
      >
        <input
          defaultChecked={false}
          id="Mythology"
          name="categorySelect"
          type="radio"
          value={20}
        />
        <span>
          Mythology
        </span>
      </label>
    </p>
    <p>
      <label
        htmlFor="Animals"
      >
        <input
          defaultChecked={false}
          id="Animals"
          name="categorySelect"
          type="radio"
          value={27}
        />
        <span>
          Animals
        </span>
      </label>
    </p>
  </fieldset>
  <fieldset
    className="options__difficulty"
  >
    <legend>
      Select a difficulty level
    </legend>
    <p>
      <label
        htmlFor="any"
      >
        <input
          defaultChecked={true}
          id="any"
          name="difficultySelect"
          type="radio"
          value="any"
        />
        <span>
          any
        </span>
      </label>
    </p>
    <p>
      <label
        htmlFor="easy"
      >
        <input
          defaultChecked={false}
          id="easy"
          name="difficultySelect"
          type="radio"
          value="easy"
        />
        <span>
          easy
        </span>
      </label>
    </p>
    <p>
      <label
        htmlFor="medium"
      >
        <input
          defaultChecked={false}
          id="medium"
          name="difficultySelect"
          type="radio"
          value="medium"
        />
        <span>
          medium
        </span>
      </label>
    </p>
    <p>
      <label
        htmlFor="hard"
      >
        <input
          defaultChecked={false}
          id="hard"
          name="difficultySelect"
          type="radio"
          value="hard"
        />
        <span>
          hard
        </span>
      </label>
    </p>
  </fieldset>
  <button
    className="btn"
    type="submit"
  >
    Submit
  </button>
</form>
`;
